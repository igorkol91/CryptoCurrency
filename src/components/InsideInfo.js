import React from 'react';
import { PropTypes } from 'prop-types';
import BackButton from './BackButton';
import image from '../image.png';

const InsideInfo = ({
  name, price, symbol, priceBtc, rank, change24,
}) => (

  <div className="dark-blue d-flex flex-column align-items-center">
    <BackButton />
    <img src={image} alt="img" className="h-25 w-25 d-block my-2 px-3" />
    <h1 className="text-white">{name}</h1>
    <span className="d-flex justify-content-between w-100 px-3">
      <h2 className="text-white">Rank:</h2>
      {' '}
      <h2 className="text-white">
        <b>
          {' '}
          {rank}
        </b>
        {' '}
      </h2>
    </span>
    <span className="d-flex justify-content-between w-100 px-3">
      <h4 className="text-white">Price: </h4>
      {' '}
      <h4 className="text-white">
        <b>
          {' '}
          {price}
        </b>
        {' '}
      </h4>
    </span>
    <span className="d-flex justify-content-between w-100 px-3">
      <h5 className="text-white">Price in BTC:</h5>
      {' '}
      <h5 className="text-white">{priceBtc}</h5>
    </span>
    <div className="text-white d-flex justify-content-between w-100 px-3">
      Change in 24 hours:
      {' '}
      <b>
        {change24}
      </b>
      {' '}
    </div>

    <div className="text-white d-flex justify-content-between w-100 px-3">
      Symbol:
      {' '}
      <b>
        {symbol}
      </b>
      {' '}
    </div>

    <div className="text-white d-flex justify-content-between w-100 px-3">
      Change in 24 hours:
      {' '}
      <b>
        {change24}
      </b>
      {' '}
    </div>
  </div>
);

InsideInfo.propTypes = {
  price: PropTypes.string.isRequired,
  priceBtc: PropTypes.string.isRequired,
  change24: PropTypes.string.isRequired,
  rank: PropTypes.number.isRequired,
  symbol: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default InsideInfo;
