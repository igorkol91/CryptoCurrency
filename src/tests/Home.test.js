import renderer from 'react-test-renderer';
import InsideInfo from '../components/InsideInfo';

describe('Home component', () => {
  test('Snapshot test', () => {
    const tree = renderer.create(
      <InsideInfo />,
    )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
