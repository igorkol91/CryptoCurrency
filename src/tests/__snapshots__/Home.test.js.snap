// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Snapshot test 1`] = `
<div
  className="dark-blue d-flex flex-column align-items-center"
>
  <div
    className="header-button align-self-start m-3"
  >
    <div
      className="back-btn"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex="0"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.67 3.87L9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
        />
      </svg>
    </div>
  </div>
  <img
    alt="img"
    className="h-25 w-25 d-block my-2 px-3"
  />
  <h1
    className="text-white"
  />
  <span
    className="d-flex justify-content-between w-100 px-3"
  >
    <h2
      className="text-white"
    >
      Rank:
    </h2>
     
    <h2
      className="text-white"
    >
      <b>
         
      </b>
       
    </h2>
  </span>
  <span
    className="d-flex justify-content-between w-100 px-3"
  >
    <h4
      className="text-white"
    >
      Price: 
    </h4>
     
    <h4
      className="text-white"
    >
      <b>
         
      </b>
       
    </h4>
  </span>
  <span
    className="d-flex justify-content-between w-100 px-3"
  >
    <h5
      className="text-white"
    >
      Circulating Supply:
    </h5>
     
    <h5
      className="text-white"
    />
  </span>
  <div
    className="text-white d-flex justify-content-between w-100 px-3"
  >
    Status:
     
    <b />
     
  </div>
</div>
`;

exports[`Navbar component Snapshot test 1`] = `
<div
  className="dark-blue d-flex flex-column align-items-center"
>
  <div
    className="header-button align-self-start m-3"
  >
    <div
      className="back-btn"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex="0"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.67 3.87L9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
        />
      </svg>
    </div>
  </div>
  <img
    alt="img"
    className="h-25 w-25 d-block my-2 px-3"
  />
  <h1
    className="text-white"
  />
  <span
    className="d-flex justify-content-between w-100 px-3"
  >
    <h2
      className="text-white"
    >
      Rank:
    </h2>
     
    <h2
      className="text-white"
    >
      <b>
         
      </b>
       
    </h2>
  </span>
  <span
    className="d-flex justify-content-between w-100 px-3"
  >
    <h4
      className="text-white"
    >
      Price: 
    </h4>
     
    <h4
      className="text-white"
    >
      <b>
         
      </b>
       
    </h4>
  </span>
  <span
    className="d-flex justify-content-between w-100 px-3"
  >
    <h5
      className="text-white"
    >
      Circulating Supply:
    </h5>
     
    <h5
      className="text-white"
    />
  </span>
  <div
    className="text-white d-flex justify-content-between w-100 px-3"
  >
    Status:
     
    <b />
     
  </div>
</div>
`;
